src/ui/public/vislib/lib/axis/axis_config.js

```diff
@@ -0,0 +1,184 @@
+import _ from 'lodash';
+import d3 from 'd3';
+export default function AxisConfigFactory() {
+
+  const defaults = {
+    show: true,
+    type: 'value',
+    elSelector: '.axis-wrapper-{pos} .axis-div',
+    position: 'left',
+    scale: {
+      type: 'linear',
+      expandLastBucket: true,
+      inverted: false,
+      setYExtents: null,
+      defaultYExtents: null,
+      min: null,
+      max: null,
+      mode: 'normal' // [percentage, normal, wiggle, silhouette]
```

Comment:
Instead of comments like these, we should store these acceptable mode values in a constants file, e.g. `scale_mode.js` and then refer to it here:

```javascript
mode: SCALE_MODE.NORMAL
```

This way the developer knows where to look to see what other kinds of values are acceptable.